include/deal.II/lac/block_vector.h

```diff
@@ -458,6 +457,54 @@ void swap (BlockVector<Number> &u,
   u.swap (v);
 }
 
+
+namespace internal
+{
+  namespace LinearOperator
+  {
+    template <typename> class ReinitRangeFactory;
+    template <typename> class ReinitDomainFactory;
+
+    /**
+     * A factory class internally used in linear_operator.h.
+     * Specialization for BlockVector<number>.
+     */
+    template<typename number>
+    class ReinitRangeFactory<BlockVector<number> >
+    {
+    public:
+      template <typename Matrix>
+      std::function<void(BlockVector<number> &, bool)>
+      operator()(const Matrix &matrix)
+      {
+        return [&matrix](BlockVector<number> &v, bool fast)
```

Comment:
`std::function` is not the problem -- this was in C++03 already and you can make it backward compatible by using `std_cxx11::function` which dispatches back to BOOST where necessary. The problem is with the use of the lambda.